src/client/content_cao.cpp

```diff
@@ -1168,14 +1168,14 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 			m_step_distance_counter = 0.0f;
 			if (!m_is_local_player && m_prop.makes_footstep_sound) {
 				const NodeDefManager *ndef = m_client->ndef();
-				v3s16 p = floatToInt(getPosition() +
-					v3f(0.0f, (m_prop.collisionbox.MinEdge.Y - 0.5f) * BS, 0.0f), BS);
+				v3pos_t p = oposToPos(getPosition() +
+					v3opos_t(0.0f, (m_prop.collisionbox.MinEdge.Y - 0.5f) * BS, 0.0f), BS);
 				MapNode n = m_env->getMap().getNode(p);
 				SimpleSoundSpec spec = ndef->get(n).sound_footstep;
 				// Reduce footstep gain, as non-local-player footsteps are
 				// somehow louder.
 				spec.gain *= 0.6f;
-				m_client->sound()->playSoundAt(spec, false, getPosition());
+				m_client->sound()->playSoundAt(spec, false, oposToV3f(getPosition()));
```

Comment:
Now what will happen when this is out of float bounds? Looks broken to me.

Reply:
float have enough bounds to handle s32 values

Reply:
> float have enough bounds to handle s32 values

Not without major precision loss. Floats only have 23 mantissa bytes.